File: src/main/kotlin/hdzi/swordsman/kotlin/_04_String.kt
package hdzi.swordsman.kotlin

/**
 * Created by taojinhou on 2018/8/20.
 */

/**
 * 模板字符串
 */
val name = "Hdzi"
fun getAge() = 20
// 用 $ 在字符串里插值,复杂插值用 ${}
val introduction = "Name: $name, age: ${getAge()}"

/**
 * 字符串字面量
 */
// 支持转移
val s = "Hello!!\n"
// 原始字符串
val text = """
    for (c in "foo"){
        println(c)
    }
""".trimIndent()

File: src/main/scala/hdzi/swordsman/scala/_04_String.scala
package hdzi.swordsman.scala

/**
  * Created by taojinhou on 2018/8/20.
  */
object _04_String {
  def main(args: Array[String]): Unit = {
    // 模板字符串
    val name = "Hdzi"

    def getAge() = 20

    // 用 $ 在字符串里插值,复杂插值用 ${}
    val introduction = s"Name: $name, age: ${getAge()}"

    // 原始字符串
    val rs =
      """
        |abc
        |def
      """.stripMargin

    // 符号
    val sym = 'abc

  }
}
